submission.py

```diff
@@ -21,17 +23,43 @@ def _catalogued_action(state, player):
 
 
 def _planned_action(game, state, player):
-    return IterativeDeepening(
-        negamax,
-        timeout=game.timeout * TIMEOUT_BUFFER,
-        max_depth=game.rows * game.columns - state.counter,
-    )(
-        game=game,
-        state=state,
-        player=player,
-        heuristic_func=heuristic,
-        order_actions_func=order_actions,
-    )
+    result = None
+    timeout = game.timeout * TIMEOUT_BUFFER
+    min_depth = 0
+    max_depth = game.rows * game.columns - state.counter
+    _LOGGER.debug(f"Setting internal timeout: {timeout}.")
+
+    async def iterative_deepening():
+        """
+        Repeats the minimax algorithm with an increasingle larger depth, and
+        saves the latest result to a nonlocal variable in the closure.
+        """
+        nonlocal result
+        for depth in range(min_depth, max_depth + 1):
+            _LOGGER.debug(f"Minimax depth: {depth}.")
+            result = await negamax(
+                game=game,
+                state=state,
+                player=player,
+                depth=depth,
+                heuristic_func=heuristic,
+                order_actions_func=order_actions,
+            )
```

Comment:
may add a debug level logging entry with result

Reply:
On it